### PR TITLE
iconbutton impl

### PR DESCRIPTION
**Pull request by edivad1999**

### CODE REVIEW

File: ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/IntUiBridge.kt [hunk 1]
```diff
@@ -900,22 +904,30 @@ private fun readEditorTabStyle(iconData: IntelliJThemeIconData, svgLoader: SvgLo
...
-    return IntUiTooltipStyle(
-        metrics = IntUiTooltipMetrics(),
-        colors = IntUiTooltipColors(
-            content = retrieveColorOrUnspecified("ToolTip.foreground"),
-            background = retrieveColorOrUnspecified("ToolTip.background"),
-            border = retrieveColorOrUnspecified("ToolTip.borderColor"),
-            shadow = Color.Black.copy(alpha = .6f),
-        ),
-    )
-}
```
Comment (rock3r) [inline ide-laf-bridge/src/main/kotlin/org/jetbrains/jewel/bridge/IntUiBridge.kt:921]: Nit: please don't do unrelated (and inconsistent with the rest of the codebase) style changes

File: core/src/main/kotlin/org/jetbrains/jewel/styling/IconButtonMetrics.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+@Stable
+interface IconButtonStyle {
+
+    val colors: IconButtonColors
+    val metrics: IconButtonMetrics
+}
+
+@Immutable
+interface IconButtonColors {
```
Comment (rock3r) [inline core/src/main/kotlin/org/jetbrains/jewel/styling/IconButtonMetrics.kt:22]: IconButtons also have a border. See `com.intellij.openapi.actionSystem.impl.IdeaActionButtonLook#paintLookBorder`. It just happens to be the same colour as the bg by default.

File: samples/ide-plugin/src/main/kotlin/org/jetbrains/jewel/samples/ideplugin/ComponentShowcaseTab.kt [hunk 3]
```diff
@@ -124,11 +125,19 @@ private fun RowScope.ColumnOne(resourceLoader: ResourceLoader) {
 
         val svgLoader = service<SwingBridgeService>().svgLoader
         Row {
-            val painterProvider = retrieveStatelessIcon("actions/close.svg", svgLoader, IntUiTheme.iconData)
+            val painterProvider = retrieveStatelessIcon("actions/more.svg", svgLoader, IntUiTheme.iconData)
```
Comment (rock3r) [inline samples/ide-plugin/src/main/kotlin/org/jetbrains/jewel/samples/ideplugin/ComponentShowcaseTab.kt:128]: Why?

File: samples/standalone/src/main/kotlin/org/jetbrains/jewel/samples/standalone/components/Buttons.kt [hunk 4]
```diff
@@ -35,5 +43,14 @@ fun Buttons() {
         DefaultButton(onClick = {}, enabled = false) {
             Text("Default disabled")
         }
+
+        IconButton(onClick = {}, modifier = Modifier.size(30.dp)) {
```
Comment (rock3r) [inline samples/standalone/src/main/kotlin/org/jetbrains/jewel/samples/standalone/components/Buttons.kt:47]: You shouldn't need to set the size explicitly — if that's required, there is some issue (e.g., missing padding)

File: core/src/main/kotlin/org/jetbrains/jewel/IconButton.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                onClick = onClick,
+                enabled = enabled,
+                role = Role.Button,
+                interactionSource = interactionSource,
+                indication = NoIndication,
+            )
+            .clip(shape)
+            .padding(style.metrics.padding)
+            .background(backgroundColor),
+        propagateMinConstraints = true,
```
Comment (devkanro) [inline core/src/main/kotlin/org/jetbrains/jewel/IconButton.kt:74]: This parameter causes the icon to be scaled to match the size of the IconButton, we don't need to scale Icon in most cases.

**Review by rock3r — APPROVED** [review]

**Pull request merged**